#160495513 Ch-mocha-test1

Johnpaul-Borree opened the pull request:
#### What does this PR do?
Test Api endpoint for create order
#### Description of Task to be completed?
- Create file to test Endpoint: Post /ap1/v1/orders
- Create tests to test for creating order taking sample order
- create tests for validating this endpoints
#### How should this be manually tested?
- Run command npm run tests 
- check continuous integrations 
- look out for readme badge on this repo
#### What are the relevant pivotal tracker stories?
#160495513

KaiserPhemi commented on .eslintrc.json:
```diff
@@ -0,0 +1,29 @@
+            "always"
+        ]
+    }
+}
```
#### Nitpick
Always add a new line after the last line

Johnpaul-Borree replied:
Ok I will do that

KaiserPhemi commented on src/controller/ordersControllers.js:
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
 
 
 export function postOrder (req, res) {
```
#### Critical
You were advised to use `ES6`. Please remove every instances of `function` keyword and use the arrow function. i.e. This should be refactored to:
```javascript
export const postOrder = (req, res) => {
  // ....
}
```

Exceptions might be configuration files.

Johnpaul-Borree replied:
Okay I will remove all instances of function keyword,
 thank you for suggesting a way

Johnpaul-Borree replied:
I have implemented the feedback on this line, thank you

KaiserPhemi commented on src/controller/ordersControllers.js:
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
+	orders.push(order);
 
-    res.json(order);
+	res.json({justAdded: order, message: 'order Created'});
```
#### Nitpick 
For objects, you should add a space between the curly braces
i.e.
```js
({ justAdded: order, message: 'order Created' })
```
Note the spaces between the contents and the curly braces. 
Also, go through your code and fix every instances of this.

Johnpaul-Borree replied:
Thanks so much, I will fix this

Johnpaul-Borree replied:
I just added a linting rule for this and corrected all appearances on this branch

KaiserPhemi commented on src/controller/ordersControllers.js:
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
+	const {error} = validateOrders(req.body);
     
-    if(error) return res.status(400).json({status: 'success', message: error.details[0].message});
+	if(error) return res.status(400).json({status: 'Failed to create order', message: error.details[0].message});
```
#### Nitpick
Please refactor this code and make it readable. It's obvious you aren't following the airbnb style guide.
For context, please check [this](https://github.com/airbnb/javascript#control-statements)

Johnpaul-Borree replied:
I just implemented this, thanks

KaiserPhemi commented on src/controller/ordersControllers.js:
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
+	res.json({justAdded: order, message: 'order Created'});
 }
 
 function validateOrders(order) {
```
#### Critical
Same thing I said above

Johnpaul-Borree replied:
Thanks, I have resolved as above

KaiserPhemi commented on src/controller/ordersControllers.js:
```diff
@@ -3,35 +3,35 @@ import orders from '../models/orders';
-    return Joi.validate(order, schema);
+	return Joi.validate(order, schema);
     
 }
```
#### Nitpick
Always add a new line after the last line of code

Johnpaul-Borree replied:
Thanks, I have corrected this

KaiserPhemi reviewed (changes requested):
Please go through the code and implement feedback.

Johnpaul-Borree committed:
Configure Eslint 

- Add eol-last script to the .eslintrc.json rule script.
- To enforce that every code block ends with a newline

Johnpaul-Borree committed:
Removing function keywords

Johnpaul-Borree committed:
Add spaces in objects in curly braces

Johnpaul-Borree committed:
Add rulefor object spaces

Johnpaul-Borree committed:
giveline space after last line

Johnpaul-Borree committed:
making code readable using Airbnb style

Johnpaul-Borree commented:
can I merge branch? @KaiserPhemi

Johnpaul-Borree committed:
Remove all function keywords

Johnpaul-Borree committed:
[ch-#160495513]implement feedback

- modified eslintrc.json to use airbnb
- removed all function keywords
- add spaces to object curly braces
- add newline at end of code block

Johnpaul-Borree committed:
[ch-#160495513] modified files for newline

Johnpaul-Borree committed:
resolved conflict in remote

Johnpaul-Borree committed:
Merge branch 'develop' into ch-mocha-test1-#160495513
